theano/gof/op.py

```diff
@@ -933,6 +961,23 @@ def make_thunk(self, node, storage_map, compute_map, no_recycling):
         # condition: either there was no c_code, or it failed
         return self.make_py_thunk(node, storage_map, compute_map, no_recycling)
 
+    def make_node(self, *inputs):
+
+        if not hasattr(self, 'itypes'):
+            raise NotImplementedError("itypes not defined")
```

Comment:
Maybe mention in the error message that the user can either define `itypes` and `otypes`, or implement `make_node`.
I would be confused it I defined `def makenode(...)` instead of def `make_node(...)` and got an error message about `itypes` instead.